### PR TITLE
Provide backup code authenticator for application login flow customization

### PR DESCRIPTION
**Pull request by chaminjay**
### Checklist
- [x] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/authenticators.tsx [hunk 1]
```diff
@@ -137,6 +137,24 @@ export const Authenticators: FunctionComponent<AuthenticatorsPropsInterface> = (
...
+            // If there exist 2FAs in the current step backup code authenticator should be enabled.
+            // Otherwise, it should be disabled.
+            if (SignInMethodUtils.countTwoFactorAuthenticatorsInCurrentStep(
+                currentStep,
+                authenticationSteps
+            ) > 0) {
+                return true;
+            } else {
+                return false;
+            }
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/authenticators.tsx:155]: WDYT think about making the code less verbose with just returning the condition, such as,

```suggestion
            return SignInMethodUtils.countTwoFactorAuthenticatorsInCurrentStep(
                currentStep,
                authenticationSteps
            ) > 0;
```

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx [hunk 2]
```diff
@@ -903,6 +962,51 @@ export const StepBasedFlow: FunctionComponent<AuthenticationFlowPropsInterface>
...
     };
 
+    /**
+     * Renders a confirmation modal to add backup code authenticator
+     * when a 2FA is added.
+     *
+     * @returns backup code authenticator add confirmation modal.
+     */
+    const renderBackupCodeAuthAddModal = (): ReactElement => (
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx:972]:
```suggestion
```

File: apps/console/src/features/applications/utils/sign-in-method-utils.ts [hunk 3]
```diff
@@ -396,9 +396,51 @@ export class SignInMethodUtils {
...
+        const hasAuthenticator: AuthenticatorInterface = authenticationSteps[currentStep].options.find(
+            (el: AuthenticatorInterface) =>
+                el.authenticator === authenticator
+        );
+
+        if (hasAuthenticator) {
+            return true;
+        } else {
+            return false;
+        }
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/utils/sign-in-method-utils.ts:442]: We can make this also less verbose, can't we?

```suggestion
        return hasAuthenticator
```

File: apps/console/src/features/identity-providers/constants/identity-provider-management-constants.ts [hunk 4]
```diff
@@ -473,7 +473,7 @@ export class IdentityProviderManagementConstants {
     public static readonly OAUTH_BEARER_AUTHENTICATOR_ID: string = "T0F1dGhSZXF1ZXN0UGF0aEF1dGhlbnRpY2F0b3I";
     public static readonly EMAIL_OTP_AUTHENTICATOR_ID: string = "ZW1haWwtb3RwLWF1dGhlbnRpY2F0b3I";
     public static readonly LEGACY_EMAIL_OTP_AUTHENTICATOR_ID: string = "RW1haWxPVFA";
-    public static readonly BACKUP_CODE_AUTHENTICATOR_ID: string = "YmFja3VwLWNvZGUtYXV0aGVudGljYXRvcgo=";
+    public static readonly BACKUP_CODE_AUTHENTICATOR_ID: string = "YmFja3VwLWNvZGUtYXV0aGVudGljYXRvcg";
```
Comment (pavinduLakshan) [inline apps/console/src/features/identity-providers/constants/identity-provider-management-constants.ts:476]: Hope this change is intentional :)

Comment (chaminjay) [reply apps/console/src/features/identity-providers/constants/identity-provider-management-constants.ts:476]: yes, the value was updated

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**